### PR TITLE
Provide console URL into PD alerts

### PR DESCRIPTION
**Pull request by cblecker**
This grabs the console URL from the system configmap (same thing `oc` does here: https://github.com/openshift/oc/blob/f2a4a0375cc7b0eacb5467a0214303983e1151d6/pkg/cli/whoami/whoami.go#L115-L130) and then puts it into the alert configuration.

This will speed up alert response time, as an SRE can click immediately from the alert to access the cluster's console.

### CODE REVIEW

**Comment by cblecker** [comment]
/hold
Holding for discussion

**Comment by cblecker** [comment]
Reviewing the test coverage report, the misses are just negative error checking. This should be safe to merge.

**Comment by cblecker** [comment]
Found an issue in testing:
Because of the way the caching client was set up, it wasn't able to grab the configmap from `openshift-config-managed`. Made some changes to allow it to read this. I don't know how I feel about them though

**Comment by cblecker** [comment]
Example of how this looks in the PD UI:
example PD alert UI

File: pkg/controller/secret/secret_controller.go
Comment (jewzaam) [inline pkg/controller/secret/secret_controller.go:411]: If we're not able to get the console URL it shouldn't block configuration of alertmanager.  We can still get stuff done, we just are not quite as efficient.  But not having alerts configured is a big problem.  Recommend we simply ignore the error and set consoleUrl to something that won't cause the config to blow up.

**Review by jewzaam — CHANGES_REQUESTED** [review]

**Comment by jewzaam** [comment]
Similar to https://github.com/openshift/configure-alertmanager-operator/pull/98 and expect some conflicts, please work with @NautiluX on how you want to merge these.

**Comment by NautiluX** [comment]
> Similar to #98 and expect some conflicts, please work with @NautiluX on how you want to merge these.

As mentioned on Slack I think both PRs provide value. Happy to rebase mine after this is merged.

**Comment by cblecker** [comment]
@jewzaam Addressed comments

**Review by NautiluX — COMMENTED** [review]
/lgtm

**Pull request merged**
